main.cpp

```diff
@@ -376,6 +463,68 @@ static int parse_conf()
     return 0;
 }
 
+static int cmpstr(const void *s1, const void *s2)
+{
+    return strcmp(*(const char **)s1, *(const char **)s2);
+}
+
+static int parse_conf_files()
+{
+    DIR *dir;
+    struct dirent *ent;
+    const char *filename, *dirname;
+    int ret;
+    char *files[128];
+    int i = 0;
+
+    // First, open default conf file
+    filename = get_conf_file_name();
+    ret = parse_conf(filename);
+
+    // If there's no default conf file, everything is good
+    if (ret < 0 && ret != -EIO) {
+        return ret;
+    }
+
+    dirname = get_conf_dir();
+    // Then, parse all files on configuration directory
+    dir = opendir(dirname);
+    if (!dir) {
+        return 0;
+    }
```

Comment:
Ah, maybe I'd have got 'from here' - sorry.
Will fix it - although any early return here leads to program termination.